### PR TITLE
Add R CMD check workflow

### LINKED ISSUE #5: Add GitHub branches strategy for contributors
@harriscw it would be good to explain this at https://pharmaverse.github.io/sdtmchecks/articles/write_a_check.html as per discussion at https://github.com/pharmaverse/sdtmchecks/pull/2, e.g. could add a simplified version of https://pharmaverse.github.io/admiraldev/devel/articles/git_usage.html

### PR DESCRIPTION
**Pull request by rossfarrugia**
Closes #5

### CODE REVIEW

**Comment by rossfarrugia** [comment]
Dinakar (@cicdguy) can you double check this basic workflow looks ok for this new pharmaverse package? we just need latest R version CMD check and no staged.dependencies.

@harriscw FYI i just went and added this for you after discussion yesterday. I also created a `devel` branch for ongoing development so you could stagger releases if you ever wished, rather than merging everything to `main` each time anything updates in your package.

File: .github/workflows/R-CMD-check.yml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        default: note
+        type: string
+  push:
+    branches:
+      - main
+      - devel
+  pull_request:
+    branches:
+      - main
+      - devel
```
[the remark this answers is not in the document]
Comment (rossfarrugia) [reply .github/workflows/R-CMD-check.yml:17]: @harriscw my thinking here is you'll want to do similar strategy to admiral contributors once you start getting external contributors getting involved right? i.e. people will make new feature branches from `devel` and then PR back to `devel`. Then you'll do a merge to `main` only for each release.

[the remark this answers is not in the document]
Comment (rossfarrugia) [reply .github/workflows/R-CMD-check.yml:17]: i extended the branch protection rule from `main` to `devel` now too assuming this is the case

**Comment by rossfarrugia** [comment]
> Looks good. Just add `knitr` to the Suggests field in the DESCRIPTION so that the vignettes can build fine.

Thanks Dinakar! was already there but just wasn't alphabetized so updated that now

**Comment by harriscw** [comment]
Thanks @cicdguy for setting this up.  The checks were failing for not being able to find `knitr`, `dplyr`, and pandoc so I managed to edit the yaml to fix this.  Is it normal that these checks seem to be fully installing several packages each run?  Run time is currently 5-6 min.

In any case it looks like now the results are just stuff that was previously expected from just running `devtools::check`, e.g. undefined global variables.  Thanks again for setting this up

**Comment by harriscw** [comment]
Thanks @cicdguy these are good suggestions

**Pull request merged**